src/analyzers/reddit_analyzer.py

```diff
@@ -25,75 +21,72 @@ def comment_length_per_subreddit(self):
             result = pd.DataFrame(list(ap.reddit_distribution_number_comments_per_user(self.collection)))
-            labels = result['min_number_of_comments']
-            bars = np.arange(len(labels))
-            values = result['number_of_users']
-
-            with _lock:
-                fig = plt.figure(figsize=(20, 15))
-                plt.bar(x=bars, height=values, align='center')
-                plt.title('Distribution of number of comments over users')
-                plt.xticks(bars, labels)
-                plt.ylabel('Number of users')
-                plt.xlabel('Number of comments')
-                plt.grid(True, axis='y')
-                st.pyplot(fig)
+            result.rename(columns={'min_number_of_comments': 'Number of comments', 'number_of_users': 'Number of users'}, inplace=True)
+            fig = px.bar(result, x='Number of comments', y='Number of users')
+            fig.update_xaxes(type='category')
+            st.write(fig)
 
     def distribution_number_posts_per_user(self):
         """
         Analyzes the distribution of posts over users (top posters vs. inactive posters)
         """
         if st.button('Show'):
             result = pd.DataFrame(list(ap.reddit_distribution_number_posts_per_user(self.collection)))
-            labels = result['min_number_of_posts']
-            bars = np.arange(len(labels))
-            values = result['number_of_users']
-
-            with _lock:
-                fig = plt.figure(figsize=(20, 15))
-                plt.bar(x=bars, height=values, align='center')
-                plt.title('Distribution of number of posts over users')
-                plt.xticks(bars, labels)
-                plt.ylabel('Number of users')
-                plt.xlabel('Number of posts')
-                plt.grid(True, axis='y')
-                st.pyplot(fig)
+            result.rename(columns={'min_number_of_posts': 'Number of posts', 'number_of_users': 'Number of users'}, inplace=True)
+            fig = px.bar(result, x='Number of posts', y='Number of users')
+            fig.update_xaxes(type='category')
+            st.write(fig)
 
     def frequently_used_news_sources(self):
         """
         Analyzes the news sources that are often used/cited on a specific subreddit
         """
+        limit = int(st.text_input("Limit", value="100"))
         subreddit = st.selectbox(label='Subreddit', options=tuple(const.SUBREDDITS))
         if st.button('Show'):
-            result = pd.DataFrame(list(ap.reddit_frequently_used_news_sources(self.collection, subreddit)))
-            st.table(result)
+            result = pd.DataFrame(list(ap.reddit_frequently_used_news_sources(self.collection, subreddit))).sort_values(by='number_of_occurrences', ascending=True)
+            result.rename(columns={'domain': 'Domain', 'number_of_occurrences': 'Number of occurrences'}, inplace=True)
+            fig = px.bar(result, x='Number of occurrences', y='Domain', orientation='h')
```

Comment:
Limit variable is introduced but never used. In line 80, change `result` to `result[:limit]`